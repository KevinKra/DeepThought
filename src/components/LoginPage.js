import React from "react";
import "../styles/_LoginPage.scss";
class LoginPage extends React.Component {
  render() {
    return (
      <main className="window">
        <section className="form-section">
          <h1>ThinkDeeper</h1>
          <form className="login">
            <label>
              {"UserName:"}
              <input type="text" placeholder="Enter your name..." />
            </label>
            <button type="submit">Log In</button>
          </form>
        </section>
      </main>
    );
  }
}

export default LoginPage;
