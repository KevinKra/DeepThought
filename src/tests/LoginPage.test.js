import React from "react";
import LoginPage from "../components/LoginPage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const defaultState = {
  renderMainPage: false
};
const mockUserName = "Squishy-big-toes";

describe("<LoginPage />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoginPage />);
  });
  it("should have a proper default state", () => {
    expect(wrapper.state()).toEqual({
      renderMainPage: false
    });
  });

  it("should render login page if renderMainPage local state is false", () => {
    expect(wrapper.state().renderMainPage).toBe(false);
    const input = wrapper.find("input").first();
    //test against snapshot while allowing input value (username) to be randomly generated
    expect(input.prop("type")).toBe("text");
    expect(input.prop("placeholder")).toBe(" Enter your name...");
  });

  it("should render main page if renderMainPage local state is true", () => {
    wrapper.setState({ renderMainPage: true });
    expect(wrapper.state().renderMainPage).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("returnToLogin method should set renderMainPage local state to false when invoked", () => {
    wrapper.setState({ renderMainPage: true });
    expect(wrapper.state()).not.toEqual(defaultState);
    wrapper.instance().returnToLogin();
    expect(wrapper.state()).toEqual(defaultState);
  });

  it("goToMainPage method should set renderMainPage local state to true when invoked", () => {
    const mockUserName = "Squishy-big-toes";
    wrapper.setState({ renderMainPage: false });
    expect(wrapper.state()).toEqual(defaultState);
    // wrapper.instance().goToMainPage({
    //   preventDefault: () => {}
    // });
    // expect(wrapper.state().renderMainPage).toEqual(true);
  });
});
