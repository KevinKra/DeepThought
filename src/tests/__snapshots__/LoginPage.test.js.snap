// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render main page if renderMainPage local state is true 1`] = `
<div>
  <MainPage
    renderLoginPage={[Function]}
  />
</div>
`;
